src/CommunityPatch/Patches/ProsperousReignPatch.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents;
+using TaleWorlds.Core;
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches {
+
+  sealed class ProsperousReignPatch : PatchBase<ProsperousReignPatch> {
+
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo TargetMethodInfo = typeof(DefaultSettlementProsperityModel).GetMethod("CalculateHearthChangeInternal", BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+
+    private static readonly MethodInfo PatchMethodInfo1 = typeof(ProsperousReignPatch).GetMethod(nameof(Prefix), BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+    private static readonly MethodInfo PatchMethodInfo2 = typeof(ProsperousReignPatch).GetMethod(nameof(Postfix), BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+
+    private PerkObject _perk;
+
+    public override void Reset()
+      => _perk = PerkObject.FindFirst(x => x.Name.GetID() == "5MjmCaUx");
+
+    public override void Apply(Game game) {
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo,
+        new HarmonyMethod(PatchMethodInfo1),
+        new HarmonyMethod(PatchMethodInfo2));
+      Applied = true;
+    }
+
+    public override bool IsApplicable(Game game) {
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo))
+        return false;
+
+      var bytes = TargetMethodInfo.GetCilBytes();
+      if (bytes == null) return false;
+      
+      var hash = bytes.GetSha256();
+      return hash.SequenceEqual(new byte[] {
+        // e.1.0.9
+        0x65, 0x10, 0x58, 0xD9, 0xC7, 0xFA, 0x99, 0x07, 
+        0xB5, 0x92, 0xC0, 0xB9, 0xC4, 0xEF, 0x94, 0xBF, 
+        0x5D, 0x7B, 0xCA, 0xA0, 0x1E, 0x37, 0xB9, 0x73, 
+        0xFA, 0x2C, 0x1D, 0x91, 0xC0, 0x5F, 0xF1, 0x14
+      });
+    }
+
+    // ReSharper disable once UnusedParameter.Local
+    private static void Prefix(Village village, ref StatExplainer explanation)
+      => explanation ??= new StatExplainer();
+
+    // ReSharper disable once InconsistentNaming
+    private static void Postfix(ref float __result, Village village, StatExplainer explanation) {
+      var perk = ActivePatch._perk;
+      if (!(village.Bound?.OwnerClan?.Leader?.GetPerkValue(perk) ?? false))
+        return;
+
+      var explainedNumber = new ExplainedNumber(__result, explanation);
+      
+      if (explanation.Lines.Count > 0)
```

Comment:
Should we also check if __result was 0, since that won't add a description to explaination if I recall correctly
(If it's something like +1, -1 we'd delete the -1)

Reply:
You mean an early return if __result is 0?

Reply:
No, new ExplainedNumber(__result, explanation) won't add an additional description if its 0:

```	if (!(value > -0.0001f) || !(value < 0.0001f))
	{
		Explainer.AddLine((baseExplanier != null) ? baseExplanier.ToString() : baseStr.ToString(), value, StatExplainer.OperationType.Base);
	}```

Reply:
I'm not following :/

Reply:
Looking at implementation of the original method don't think its possible anyways but i'll keep going haha.

       `explanation.Lines.RemoveAt(explanation.Lines.Count - 1);`

^ That line could get executed despite no 'Base +n' getting added

Reply:
Oh wait maybe i'm being dumb.

Reply:
I see now. So `if (__result != 0 && explantion.Lines.Count > 0)`?

Reply:
Yea, though `(!(value > -0.0001f) || !(value < 0.0001f))` if we want to follow their logic to not add a line

Reply:
Check was added, resolving this convo.